feat(plugin): add bob launchpad to icp package.

bob-robert-ai opened the pull request:
Add https://launch.bob.fun, the most popular launchpad platform on ICP to the ICP package.

# Relates to:

Following this PR: https://github.com/ai16z/eliza/pull/357

# Risks

Low

odilitime commented on packages/plugin-icp/src/actions/createToken.ts:
```diff
@@ -139,7 +167,7 @@ export const executeCreateToken: Action = {
                 twitter: response.twitter,
                 telegram: response.telegram,
-            });
+            }, 'launch-bob');
```
why is this hard coded?

bob-robert-ai replied:
launch.bob.fun is the most widely used launchpad on ICP, so it felt logical to set it as the default.

odilitime commented on packages/plugin-icp/src/actions/createToken.ts:
```diff
@@ -65,12 +98,7 @@ async function createTokenTransaction(
-            twitter: unwrapOption(ok.twitter),
-            website: unwrapOption(ok.website),
-            telegram: unwrapOption(ok.telegram),
-        }),
```
why are these taken out in this branch?

bob-robert-ai replied:
Added back.

lalalune commented:
@bob-robert-ai there are some conflicts with some other ICP-related create token functionality, would you mind reviewing that?

bob-robert-ai commented:
> @bob-robert-ai there are some conflicts with some other ICP-related create token functionality, would you mind reviewing that?

Just fixed the conflicts, could you take a look?

Commit:
feat: support for eternalai provider can write request/response

Commit:
feat: support for eternalai provider can write request/response

Commit:
feat: support for eternalai provider can write request/response

sin-bufan committed:
Merge branch 'develop' into fix-twitter-actionInterval

AIFlowML committed:
fix(client-slack): implement Media type properties in message attachments - #1384

Commit:
Merge branch 'develop' into eternal-ai-pr

AIFlowML committed:
fix(postgres): Handle vector extension creation properly (#1561)

- Add proper checks for vector extension existence

- Create extensions schema if not exists

- Add extensions schema to search path

- Ensure idempotent schema creation

sinecose committed:
feat(near): replace console.log to elizaLogger

sinecose committed:
feat(near/transfer): replace console.log to elizaLogger

sinecose committed:
feat(near/wallet): replace console.log to elizaLogger

TresFlames committed:
Update faq.md

Removed mentions of some items.

tcm390 committed:
use 4o as default model

tcm390 committed:
Merge branch 'develop' into tcm-trimTokens

tcm390 committed:
use elizaLogger

shakkernerd committed:
Merge branch 'develop' into Docs-Patch-1

shakkernerd committed:
Merge pull request #1746 from elizaOS/Docs-Patch-1

docs: update faq.md

marcellodesales committed:
Merge branch 'develop' into feature/dockerized-docs-support

shakkernerd committed:
Merge branch 'develop' into kor.md-modify

shakkernerd committed:
Merge pull request #1739 from osrm/kor.md-modify

docs: update README_KOR.md

marsic3 committed:
feat: add Serbian README translation and enhance main README with new features and setup instructions

tcm390 committed:
move trimTokens to core

shakkernerd committed:
Merge branch 'develop' into marsic3/add-serbian

shakkernerd committed:
Merge pull request #1757 from marsic3/marsic3/add-serbian

docs: add Serbian README translation

shakkernerd committed:
Merge branch 'develop' into fix/lint_errors

ryanleecode committed:
Merge remote-tracking branch 'origin/develop' into feat/live-monorepo-types

tcm390 committed:
clean code

marcellodesales committed:
Merge branch 'develop' into feature/dockerized-docs-support

tcm390 committed:
restore test

tcm390 committed:
clean code

tcm390 committed:
remove tokenizer service

gmh5225 committed:
fix(client): improve Windows compatibility for Vite dev server

- Add host configuration to allow IP access
- Use explicit IP address instead of localhost
- Support dynamic server port from environment variables

tcm390 committed:
fall back if unsupported type

tcm390 committed:
Merge branch 'develop' into tcm-trimTokens

odilitime committed:
Merge branch 'develop' into fix/slack-1384-media-type-attachments

tcm390 committed:
Move encoding into try block to handle potential errors during model encoding.

gmh5225 committed:
Merge branch 'develop' into fix-windows

odilitime committed:
Merge branch 'develop' into fix/postgres-1561-vector-extension

KacperKoza343 committed:
chore: update readme

shakkernerd committed:
feat: add JsDoc to trimTokens function

shakkernerd committed:
feat: add validation to trimTokens

shakkernerd committed:
Merge pull request #1605 from elizaOS/tcm-trimTokens

fix: Support for Non-OpenAI Models in Token Trimming

ryanleecode committed:
Merge branch 'develop' into feat/live-monorepo-types

KacperKoza343 committed:
chore: remove character from Development section in README

KacperKoza343 committed:
chore: update README

shakkernerd committed:
Merge branch 'develop' into fix/lint_errors

shakkernerd committed:
Merge pull request #1759 from elizaOS/fix/lint_errors

fix: build lint errors

KacperKoza343 committed:
Merge pull request #10 from blockydevs/update/cosmos-plugin-readme

chore: update readme

shakkernerd committed:
Merge branch 'develop' into fix-windows

shakkernerd committed:
Merge pull request #1760 from gmh5225/fix-windows

fix(client): improve Windows compatibility for Vite dev server

shakkernerd committed:
Merge branch 'develop' into feat/live-monorepo-types

shakkernerd committed:
Merge pull request #1365 from ryanleecode/feat/live-monorepo-types

feat: use custom conditions for live monorepo types

shakkernerd committed:
Merge remote-tracking branch 'origin/develop' into pr/genesis-0000/1740

shakkernerd committed:
fix: handle env using runtime getSetting function

pedronovikovborges committed:
docs: add Romanian README translation

mcp97 committed:
Merge branch 'develop' into console-to-elizaLogger

mcp97 committed:
Merge pull request #1745 from sinecose/console-to-elizaLogger

feat(plugin-near): replace console.log to eliza logger

dxlliv committed:
Merge branch 'develop' into main

ai16z-demirix committed:
refactor: (draft) improving API error handling for coinbase integration

ai16z-demirix committed:
Merge branch 'develop' into main

ai16z-demirix committed:
test: adding more improved tests for providers, memory and goals.

augchan42 committed:
fix: remove twitter profile caching (#1638)

madjin committed:
Merge branch 'develop' into feature/full-plugin-docs

madjin committed:
Merge pull request #1675 from Ed-Marcavage/feature/full-plugin-docs

Agentic Eliza Plugin Documenter - Multilingual (e.g., English, Spanish, French)

swizzmagik committed:
fix: line break handling in chat

madjin committed:
Merge branch 'develop' into feature/dockerized-docs-support

madjin committed:
Merge pull request #1722 from marcellodesales/feature/dockerized-docs-support

🐛 fix plugins.md formatting for docs with dockerized docs validation

shakkernerd committed:
Merge branch 'develop' into fix/line-break-formatting-in-chat

shakkernerd committed:
Merge pull request #1784 from swizzmagik/fix/line-break-formatting-in-chat

fix: line break handling in chat

RobertSloan22 committed:
Create README.md

Adding README.md for the client-github package

Rudrakc committed:
Removed FerePro plugin

shakkernerd committed:
chore: remove unused import and var

shakkernerd committed:
Merge branch 'develop' into fix/lint_errors

shakkernerd committed:
Merge pull request #1797 from elizaOS/fix/lint_errors

chore: remove unused import and var

marcellodesales committed:
:bug: generating package-lock.json for docusaurus updates

marcellodesales committed:
:whale: Just changing Dockerfile.docs with build comments

marcellodesales committed:
:scroll: update package-lock.json with current docs package.json

marcellodesales committed:
:scroll: update pnpm-lock.yaml with docusaurus changes

mcp97 committed:
Merge branch 'develop' into fix-twitter-actionInterval

mcp97 committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge branch 'develop' into pr-1795

shakkernerd committed:
Merge branch 'develop' into fix/slack-1384-media-type-attachments

shakkernerd committed:
Merge pull request #1741 from AIFlowML/fix/slack-1384-media-type-attachments

fix(client-slack): implement Media type properties in message attachments #1384

mcp97 committed:
Merge branch 'develop' into fix-twitter-actionInterval

shakkernerd committed:
Merge branch 'develop' into fix/postgres-1561-vector-extension

shakkernerd committed:
Merge pull request #1743 from AIFlowML/fix/postgres-1561-vector-extension

fix(postgres): Handle vector extension creation properly (#1561)

shakkernerd committed:
Revert "fix(postgres): Handle vector extension creation properly (#1561)"

shakkernerd committed:
Merge pull request #1799 from elizaOS/revert-1743-fix/postgres-1561-vector-extension

Revert "fix(postgres): Handle vector extension creation properly (#1561)"

shakkernerd committed:
Merge branch 'develop' into fix-twitter-actionInterval

shakkernerd committed:
Merge pull request #1738 from sin-bufan/fix-twitter-actionInterval

fix: standardize ACTION_INTERVAL unit to minutes in twitter client

shakkernerd committed:
Merge branch 'develop' into refactor/client-react-query

shakkernerd committed:
Merge pull request #1713 from aalimsahin/refactor/client-react-query

refactor: client api

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #1712 from dxlliv/main

feat: Simulate discord typing while generating a response

shakkernerd committed:
Merge branch 'develop' into pr-1616

augchan42 committed:
feat(twitter): allow leading numbers and wildcard in target usernames

Fix for #1556
- Added support for Twitter usernames starting with numbers
- Added support for wildcard '*' in TWITTER_TARGET_USERS
- Fixed test discovery in test.sh to find all .test.ts files
- Updated Jest config for better ESM support

The main change allows Twitter target usernames to start with numbers and
use wildcards, matching current Twitter/X username rules. Test infrastructure
was improved to properly handle ESM modules and find all test files.

Changes:
- Updated username validation regex in Twitter client
- Added test cases for numeric usernames and wildcards
- Switched test.sh to use find for reliable test discovery
- Simplified Jest config to focus on ESM support

tcm390 committed:
refactor model structure

tcm390 committed:
refactor model structure

tcm390 committed:
add getModelSettings function

augchan42 committed:
Merge branch 'develop' into fix/twitter-wildcard-targets-1556

tcm390 committed:
clean code

augchan42 committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

shakkernerd committed:
chore: install with no frozen-lockfile flag

shakkernerd committed:
Merge pull request #1802 from elizaOS/shakkernerd-patch-1

chore: install with no frozen-lockfile flag

tcm390 committed:
seperate getImageModelSettings and getModelSettings

shakkernerd committed:
fix: generatioon tests for trimTokens

shakkernerd committed:
Merge pull request #1803 from elizaOS/fix/tests

fix: generation tests for trimTokens

tcm390 committed:
typo

tcm390 committed:
clean code

shakkernerd committed:
chore: bump version to v.0.1.7

tcm390 committed:
add return type of function

shakkernerd committed:
Merge pull request #1804 from elizaOS/new_version_0.1.7

chore: bump version to v.0.1.7

shakkernerd committed:
Merge pull request #1717 from elizaOS/develop

chore: Develop => main for 1.7.0 release

tcm390 committed:
adapt to new model structure

shakkernerd committed:
chore: install with no frozen-lockfile flag

actions-user committed:
chore: update changelog

jazzvaz committed:
add .pnpm-store to gitignore

jazzvaz committed:
add devcontainer

Commit:
feat(models): update Google model configurations

- Add gemini-2.0-flash-exp for SMALL/MEDIUM class
- Add gemini-exp-1206 for LARGE class
- Update default models to latest versions

gmh5225 committed:
refactor(models): replace gemini-exp-1206 with gemini-2.0-flash-exp

gmh5225 committed:
replace gemini-exp-1206 with gemini-2.0-flash-exp

macfly-base committed:
Update c3po.character.json

Changes & Optimizations:

Formatting Consistency:
Indented uniformly for readability.
Grouped related sections logically for better navigation.

Removed Redundancy
Simplified the structure in messageExamples to avoid unnecessary repetition.

Clarity in twitterSpaces:
Combined related attributes like enableIdleMonitor and enableRecording for a cohesive view of space features.

Future-Proof:
Added structure for easy expansion (e.g., topics and knowledge).
This ensures better maintainability while keeping it functional and user-friendly.

Commit:
refactor: typo in readme

migace committed:
fix(): env file, remove chagnes from defaultCharacter file and revert package.json

rawgroundbeef committed:
fix(solana token provider): await cached data

bitcoinbender committed:
Update .env.example

Commit:
fix: add default export to plugin-image-generation

mcp97 committed:
Merge pull request #1831 from jonathanmv/develop

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1829 from bitcoinbender/patch-1

mcp97 committed:
Merge branch 'develop' into refactor/readme

affaan-m committed:
chore: update bootstrap plugin export

Relates to:
Plugin export collisions in character files

Risks
Low - This is a minor bug fix that ensures proper module exports

Background
What does this PR do?
Ensures the bootstrap plugin has a proper default export at the bottom to avoid collisions when used in the actions module in character files. This maintains consistency with other plugins in the codebase that follow the same pattern (like the EVM plugin).

What kind of change is this?
Bug fixes (non-breaking change which fixes an issue)

Documentation changes needed?
My changes do not require a change to the project documentation.

Testing
Where should a reviewer start?
Check packages/plugin-bootstrap/src/index.ts to verify the proper export structure
Verify that the plugin can be properly imported in character files without collisions
Detailed testing steps
Load a character file that uses the bootstrap plugin
Verify that no export collisions occur when the plugin is loaded
Verify that all plugin actions are properly accessible
Deploy Notes
No special deploy steps required. This is a straightforward bug fix that maintains backward compatibility.

Discord:
affaanmustafa

mcp97 committed:
Merge branch 'develop' into develop

ai16z-demirix committed:
improvement: replacing console.log with elizaLogger.

ai16z-demirix committed:
Merge branch 'develop' into improvement/eliza-logger

web3isthefuture committed:
Update README_KOR.md

Modified the previous sentence to align with the meaning in Korean.

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1815 from gmh5225/develop

mcp97 committed:
Merge branch 'develop' into improvement/eliza-logger

mcp97 committed:
Merge pull request #1834 from ai16z-demirix/improvement/eliza-logger

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge branch 'develop' into refactor/readme

ai16z-demirix committed:
Revert "refactor: (draft) improving API error handling for coinbase integration"

This reverts commit 3a9e84e538b9253d567aa32bde3bb9d7b67ba274.

Commit:
correct faq docs link in both contribution md files

mcp97 committed:
Merge branch 'develop' into patch-4

ai16z-demirix committed:
test: fixing generation.test.ts. Adding tests for memory, goals and provider

ai16z-demirix committed:
Merge branch 'develop' into test/goals-memory-provider

MacsDickinson committed:
Merge branch 'develop' into 1838--fix-faq-link-in-contributing-md

poodengDev committed:
Fix faq link on contributing.md

poodengDev committed:
Update contributing link aswell to faq

poodengDev committed:
fix broken links

shakkernerd committed:
Merge branch 'develop' into 1838-fix-faq-link-mtoor

Gajesh2007 committed:
fix: fixed the anthropic regex

Gajesh2007 committed:
Merge branch 'main' of https://github.com/Gajesh2007/eliza

Gajesh2007 committed:
Merge main: Add reclaim adapter and resolve conflicts

Gajesh2007 committed:
fix: conflicts

Gajesh2007 committed:
Merge branch 'develop' into main

aisjdoaisjdoi2123 committed:
Update .env.example

lalalune committed:
Merge pull request #1835 from IterateFast-Labs/develop

Update README_KOR.md

lalalune committed:
Merge pull request #1832 from Valetek/refactor/readme

refactor: typo in readme

lalalune committed:
Merge pull request #1847 from manotoor/1838-fix-faq-link-mtoor

Fix faq link on contributing.md

lalalune committed:
Merge pull request #1787 from RobertSloan22/patch-1

Create README.md

lalalune committed:
Merge pull request #1827 from macfly-base/main

Update c3po.character.json

mcp97 committed:
Merge branch 'develop' into test/goals-memory-provider

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Merge branch 'develop' into develop

lalalune committed:
Merge branch 'feat/genlayer-plugin' of https://github.com/yeagerai/eliza into HEAD

daniel-farina committed:
Add Stargaze plugin

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'bugfix/broken-ci-docs-missed-frozen' of https://github.com/marcellodesales/elizaOS into HEAD

lalalune committed:
Merge branch 'patch-4' of https://github.com/affaan-m/eliza into HEAD

lalalune committed:
Merge branch '1838--fix-faq-link-in-contributing-md' of https://github.com/MacsDickinson/eliza into HEAD

lalalune committed:
Merge branch 'develop' of https://github.com/inferanetwork/eliza into HEAD

lalalune committed:
Merge branch 'main' of https://github.com/robin-rrt/eliza into HEAD

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'develop' of https://github.com/blockydevs/eliza into HEAD

lalalune committed:
Merge branch 'roshan/plugin-tee-marlin' of https://github.com/marlinprotocol/eliza into HEAD

lalalune committed:
fix lock

odilitime committed:
Add DAO donation ask & dev discord

odilitime committed:
Merge pull request #1867 from elizaOS/odilitime-patch-1

docs: Add DAO donation ask & dev discord

tcm390 committed:
remove unused variable

tcm390 committed:
remove unused variable

tcm390 committed:
add EmbeddingModelSettings type

tcm390 committed:
get the info from model settings instead of hardcoced number

tcm390 committed:
remove unused variable

tcm390 committed:
fix provider path

tcm390 committed:
Merge branch 'main' into tcm-refactor-modelConfig

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
fix bad merge

odilitime committed:
revert 1834 changes

odilitime committed:
audit fix

shakkernerd committed:
Merge pull request #1872 from elizaOS/fix-postinstall-script

fix: Fix postinstall script

shakkernerd committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
Merge pull request #1770 from pedronovikovborges/docs/readme-ro

docs: add Romanian README translation

tcm390 committed:
resolve conflict

shakkernerd committed:
Merge branch 'develop' into test/goals-memory-provider

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

shakkernerd committed:
Merge pull request #1840 from ai16z-demirix/test/goals-memory-provider

test: adding more tests for goals,memory and provider. Fixing generation.test.ts

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

odilitime committed:
Merge pull request #1782 from augchan42/fix/twitter-profile-cache-1638

fix: remove twitter profile caching (#1638)

shakkernerd committed:
Merge branch 'develop' into add-devcontainer

shakkernerd committed:
Merge pull request #1807 from jazzvaz/add-devcontainer

feat: Add devcontainer

shakkernerd committed:
Merge branch 'develop' into fix/await-cached-prices-2

shakkernerd committed:
Merge pull request #1828 from bentatum/fix/await-cached-prices-2

fix(solana token provider): await cached data

odilitime committed:
support more debians distros

odilitime committed:
Merge branch 'develop' into fix-postinstall-script

shakkernerd committed:
Merge pull request #1875 from odilitime/fix-postinstall-script

chore: support more debians distros

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

shakkernerd committed:
Merge pull request #1805 from elizaOS/tcm-refactor-modelConfig

feat: refactor model config

rawgroundbeef committed:
chore: enable strict null checks

ai16z-demirix committed:
test: adding test setup for telegram client

ai16z-demirix committed:
test: adding test setup for telegram client - improving message manager tests

ai16z-demirix committed:
Merge branch 'develop' into test/client-telegram

kylebuildsstuff committed:
Add OpenWeather plugin

Arrowana committed:
Use recommended settings in jupiter swap

JoeyKhd committed:
upd: fixed broken API Documentation URL

JoeyKhd committed:
upd: fixed path

Gajesh2007 committed:
Merge branch 'elizaOS:main' into main

Gajesh2007 committed:
Merge old into develop with regenerated pnpm-lock.yaml

Gajesh2007 committed:
Merge develop into main

Gajesh2007 committed:
fix: settings error

fabianhug committed:
Merge pull request #1558 from Gajesh2007/main

feat: added new plugin - zktls - reclaim

augchan42 committed:
Merge branch 'elizaOS:main' into fix/twitter-wildcard-targets-1556

HashWarlock committed:
feat: add remote attestation action

augchan42 committed:
just add needed change to run unit test

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge branch 'develop' into pr-1884

shakkernerd committed:
chore: fix broken pnpm lock file

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: change file extension from rs -> ts

shakkernerd committed:
Merge branch 'develop' into pr-1885

shakkernerd committed:
Merge branch 'develop' into chore/jupiter-swap-recommended-settings

shakkernerd committed:
Merge pull request #1882 from Arrowana/chore/jupiter-swap-recommended-settings

feat: Use recommended settings in jupiter swap

shakkernerd committed:
Merge branch 'develop' into fix-broken-docs-api-url

shakkernerd committed:
Merge pull request #1881 from JoeyKhd/fix-broken-docs-api-url

Fixed broken API Documentation URL

shakkernerd committed:
Merge branch 'develop' into pr-1880

shakkernerd committed:
Merge branch 'develop' into test/client-telegram

shakkernerd committed:
Merge pull request #1879 from ai16z-demirix/test/client-telegram

test: adding test setup for telegram client

shakkernerd committed:
Merge branch 'develop' into pr-1878

shakkernerd committed:
fix: conflicting Readme.md files

zhourunlai committed:
fix case sensitive path

shakkernerd committed:
Merge branch 'develop' into fix_case_sensitive_path

shakkernerd committed:
Merge pull request #1892 from zhourunlai/fix_case_sensitive_path

fix: remove case sensitive path

odilitime committed:
hyperfi endpoint, clean up multer/imports/console logs, notes

shakkernerd committed:
Merge pull request #1896 from odilitime/hfi

feat: hyperfy support

wtfsayo committed:
test commit

wtfsayo committed:
Merge branch 'develop' into pr/714

wtfsayo committed:
fix: case-sensitive readme issue

wtfsayo reviewed (approved):
~~Low risk, ready to merge, integration test failing because `develop` is broken~~

wtfsayo reviewed (changes requested):
Did a more thorough review. looks like competition between PickPump and LaunchBob... please submit a proper PR than trying to force replace yours!

bob-robert-ai commented:
How do you usually go about competing launchpads?
